apps/studio/src/server/modules/version/version.service.ts

```diff
@@ -0,0 +1,43 @@
+import type { SelectExpression, Transaction } from "kysely"
+import { type DB } from "~prisma/generated/generatedTypes"
+
+import { db } from "../database"
+
+const defaultVersionSelect: SelectExpression<DB, "Version">[] = [
+  "Version.id",
+  "Version.versionNum",
+  "Version.resourceId",
+  "Version.blobId",
+  "Version.publishedAt",
+]
+
+export const getVersionById = ({ versionId }: { versionId: string }) =>
+  db
+    .selectFrom("Version")
+    .where("Version.id", "=", versionId)
+    .select(defaultVersionSelect)
+    .executeTakeFirstOrThrow()
+
+export const createVersion = async (
+  props: {
+    versionNum: number
+    resourceId: number
+    blobId: number
+  },
+  tx?: Transaction<DB>,
+) => {
+  const { versionNum, resourceId, blobId } = props
+  const instance = tx ? tx : db
+  const addedVersion = await instance
+    .insertInto("Version")
+    .values({
+      versionNum: String(versionNum),
```

Comment:
issue: already mentioned in the other comments but I think that this might potentially be dangerous and opens us up to race conditions, especially when the version number from 2 different requests are the same.

Reply:
I have added serializable isolation levels